src/server/pps/server/worker_pool.go

```diff
@@ -177,16 +186,17 @@ func (w *workerPool) discoverWorkers(ctx context.Context) {
 }
 
 func (w *workerPool) addWorker(addr string) error {
-	if cancel, ok := w.workersMap[addr]; ok {
-		cancel()
+	w.workersMapMu.RLock()
```

Comment:
Do we need this mutex?  I think `addWorker` and `delWorker` are only ever called from the discovery goroutine.